doc/opam-provided.mld

```diff
@@ -0,0 +1,105 @@
+{1 [opam]-provided dependencies}
+
+This section documents a feature that’s useful when some dependencies cannot be
+installed via [opam-monorepo]. This workflow isn’t meant for regular use but
+only as a way to use [opam-monorepo] in cases where it wouldn't be
+otherwise possible.
+
+We suggest minimizing its usage and prefer to use (and contribute) [dune-ports]
+from the [opam-overlays] repository. It's usage is safest with
+leaf-dependencies and those that don’t have dependency intersections with
+vendored packages. Due to possibly of unexpected interactions, this feature is only
+meant for advanced users, and the solutions it produces might be in flux.
+
+{2 Usecases}
+
+Sometimes it is not possible to put all your dependencies into the [duniverse],
+be it due to your dependencies not building with [dune] or some tooling that
+should be just installed via [opam]. Such examples include:
+
+{ul
+{- Packages not building with Dune}
+{- Packages that should not be vendored for legal reasons}
+{- Packages that just provide helper binaries like [utop], [ocamlformat]}
```

Comment:
I don't think this is a very good example as ocamlformat and possibly utop as well can be vendored seemlessly. It's probably best to replace this with concrete example we know will use this such as `ocaml-freestanding` or foreign language libraries such as tezos-rust-libs for instance.

Reply:
I've removed it for now, we can add it later if we find good examples. `tezos-rust-libs` feels a bit too obscure to add, maybe `ocaml-freestanding` indeed.